rest/blip_client_test.go

```diff
@@ -1456,19 +1325,42 @@ func (btcc *BlipTesterCollectionClient) WaitForVersion(docID string, docVersion
 	return data
 }
 
+// getBody returns the body for a specific revision. This will fail the test harness if not present.
+func (btcc *BlipTesterCollectionClient) getBody(docID string, version DocVersion) db.Body {
+	btcc.seqLock.RLock()
+	defer btcc.seqLock.RUnlock()
+	doc, ok := btcc._getClientDoc(docID)
+	require.True(btcc.TB(), ok, "docID %q not found", docID)
+	var body db.Body
+	require.NoError(btcc.TB(), body.Unmarshal(doc._getRev(btcc.TB(), version).body))
+	return body
+}
+
+// getBodyAndDeltas returns the body and delta from version to serverVersion. If serverVersion was a tombstone, do not return delta. This will fail the test harness if not present.
+func (btcc *BlipTesterCollectionClient) getBodyAndDeltas(docID string, version DocVersion, serverVersion DocVersion) (body []byte, deltaBody db.Body, parentVersion *DocVersion) {
```

Comment:
This function isn't returning a delta - though it arguably should. It is only returning a full server version body right now.

We need to also move the `Diff()` code down into here to to achieve that.

Reply:
Hopefully I made this code more clear now.